src/parser.rs

```diff
@@ -975,7 +975,7 @@ impl<'a> Parser<'a> {
     /// parse a group by expr. a group by expr can be one of group sets, roll up, cube, or simple
     /// expr.
     fn parse_group_by_expr(&mut self) -> Result<Expr, ParserError> {
-        if dialect_of!(self is PostgreSqlDialect | DuckDbDialect | GenericDialect) {
+        if self.dialect.supports_filter_during_aggregation() {
```

Comment:
I think this should be:

```suggestion
        if self.dialect.supports_group_by_expr() {
```

Reply:
Thanks for catching that Andy. Seems my autocomplete got a little excited there.